src/login/tools.js

```diff
@@ -0,0 +1,261 @@
+import _ from 'underscore';
+
+import {staticAttributePatterns, otherAttributePatterns} from './consts';
+import {filters} from './filters';
+
+/**
+ * Iterates all inputs in an input cluster testing them against provided filter function with optional extra arguments for testing patterns
+ * @param {Object[]} inputCluster
+ * @param {Function} filter
+ * @param {Object} extras
+ * @returns {Object | *}
+ */
+const reduceInputs = ({inputCluster, filter}, extras = {}) =>
+    inputCluster.reduce((cache, input, index, array) => {
+        console.log('testing:', _.pick(input, value => typeof value === 'string'));
```

Comment:
Use `log.info` instead of `console` everywhere.

Reply:
Done and then reversed back again as the SDK log function significantly reduces readability for debugging due to merging everything on one line. This kind of destroys the structure of the log messages which would lose their purpose - probably can be solved with some stringification options for objects - will address this at a later stage.

Reply:
Not sure what you mean. You can pass `data` as the second parameter to `log.info(message, data)` and it will print the object quite nicely.

Reply:
Cool, didn't know that! Will look into it again.

Reply:
I have tried switching to the SDK logger but always keep coming back for the reasons below:

- I like to log plain objects because of their inherent structure and good default wrapping and readability. I understand the need to enforce a single line to condense logs for long runs and scrapers but it's far from ideal for any nested objects with more than a couple of properties readability-wise and scanning a single line to spot something is really tough on the eyes.

- `data` are printed really dark but default and almost illegible in the console (maybe easy fix)

- `data` forced as second param whereas with default logger we can just chain things forever without thinking about the order `console.log(1,2,3,4,5)` so basically instead of this:

`console.log('patterns', patterns);`
we're forced into this in the worst case: (or template literals)
`log.info('stringifiedObject' + JSON.stringify(JSON.parse(stringifiedObject), null, 2));`

another disadvantage being that instead of this:

```
patterns {
  username: [
    { attribute: 'type', value: 'email' },
...
```
wrapping them becomes overkill and spans over entire screen:
```
INFO  patterns {
  "username": [
    {
      "attribute": "type",
      "value": "email"
    },
......................................
```